src/main/scala/julienrf/json/derived/FieldAdapter.scala

```diff
@@ -0,0 +1,25 @@
+package julienrf.json.derived
+
+trait Adapter extends (String => String)
+
+object Adapter {
+
+  val snakeCase = Adapter {
+    (s: String) => {
+      s.foldLeft(new StringBuilder) {
+        case (s, c) if Character.isUpperCase(c) && s.nonEmpty => s append "_" append (Character toLowerCase c)
+        case (s, c) => s append c
+      }.toString
+    }
+  }
+
+  val identity = Adapter(s => s)
+
+  val default = identity
```

Comment:
It is not necessary to have both `identity` and `default`.

Also, can you add scaladoc explaining what each `Adapter` does?